src/project/mod.rs

```diff
@@ -549,6 +549,40 @@ impl Project {
         Ok(())
     }
 
+    pub fn remove_dependency(
+        &mut self,
+        name: impl AsRef<str>,
+        platform: Option<Platform>,
+    ) -> miette::Result<()> {
+        let removed = match platform {
+            Some(p) => {
+                let t = ensure_toml_target_table(&mut self.doc, p, "dependencies")?;
```

Comment:
This function has the side effect that if the table doesn't exist it is added. Even though the dependency doesn't get removed if it doesn't exist you still end up with a modified document.